### PR TITLE
[1.3] bucket.Put: copy key before seek

### PR DESCRIPTION
**Pull request by fuweid**
cherry-pick: 
- https://github.com/etcd-io/bbolt/pull/637
- https://github.com/etcd-io/bbolt/pull/550
- https://github.com/etcd-io/bbolt/pull/641

### CODE REVIEW

File: bucket.go [hunk 1]
```diff
@@ -288,18 +288,23 @@ func (b *Bucket) Put(key []byte, value []byte) error {
 		return ErrValueTooLarge
 	}
 
+	// Insert into node.
```
Comment (fuweid) [inline bucket.go:291]: It's related to https://github.com/etcd-io/bbolt/pull/550.

If it's not accepted, I can use `key` here. Or I can cherry-pick 550 first.

Comment (ahrtr) [reply bucket.go:291]: Please backport both https://github.com/etcd-io/bbolt/pull/550 and https://github.com/etcd-io/bbolt/pull/637. They are highly related, so let's just backport both of them in this PR.

Comment (fuweid) [reply bucket.go:293]: Updated

File: bucket.go [hunk 2]
```diff
@@ -183,15 +183,17 @@ func (b *Bucket) CreateBucket(key []byte) (*Bucket, error) {
 	var value = bucket.write()
 
 	// Insert into node.
-	key = cloneBytes(key)
-	c.node().put(key, key, value, 0, bucketLeafFlag)
+	// Tip: Use a new variable `newKey` instead of reusing the existing `key` to prevent
+	// it from being marked as leaking, and accordingly cannot be allocated on stack.
+	newKey := cloneBytes(key)
```
Comment (ahrtr) [inline bucket.go:188]: I think we should clone the key before `seek` (line 167). Please update the main branch firstly. Thanks.

**Review by ahrtr — APPROVED** [review]
lgtm

Thanks

**Pull request merged**